jerry-core/ecma/builtin-objects/ecma-builtin-json.cpp

```diff
@@ -1115,28 +1115,21 @@ ecma_builtin_json_stringify (ecma_value_t this_arg __attr_unused___, /**< 'this'
     }
     else
     {
-      ecma_length_t string_len = ecma_string_get_length (space_str_p);
+      ecma_length_t string_size = ecma_string_get_size (space_str_p);
 
-      MEM_DEFINE_LOCAL_ARRAY (zt_string_buff, string_len, lit_utf8_byte_t);
+      MEM_DEFINE_LOCAL_ARRAY (string_buff, string_size, lit_utf8_byte_t);
 
-      size_t string_buf_size = (size_t) (string_len) * sizeof (lit_utf8_byte_t);
       ssize_t bytes_copied = ecma_string_to_utf8_string (space_str_p,
-                                                         zt_string_buff,
-                                                         (ssize_t) string_buf_size);
+                                                         string_buff,
+                                                         (ssize_t) string_size);
       JERRY_ASSERT (bytes_copied > 0);
 
-      /* Buffer for the first 10 characters. */
-      MEM_DEFINE_LOCAL_ARRAY (space_buff, 10, lit_utf8_byte_t);
+      lit_utf8_iterator_t iter = lit_utf8_iterator_create (string_buff, string_size);
+      lit_utf8_iterator_advance (&iter, 10);
 
-      for (uint32_t i = 0; i < 10; i++)
-      {
-        space_buff[i] = zt_string_buff[i];
-      }
+      context_p.gap_str_p = ecma_new_ecma_string_from_utf8 (string_buff, iter.buf_pos.offset);
```

Comment:
And one more issue, `context_p` is not a pointer, could you please rename it to `context`?